core/src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java

```diff
@@ -102,11 +102,15 @@ public void collect(int parentDoc, long bucket) throws IOException {
                 }
             };
         } else {
-            doPostCollection();
             return bufferingNestedLeafBucketCollector = new BufferingNestedLeafBucketCollector(sub, parentDocs, childDocs);
         }
     }
 
+    @Override
+    protected void preGetChildLeafCollectors() throws IOException {
+        doPostCollection();
```

Comment:
nit: It seems weird to call `doPostCollection()` during collection. Maybe we should have a separate method that both call?